pkg/operator/revisioncontroller/revision_controller.go

```diff
@@ -366,6 +366,21 @@ func (c RevisionController) sync(ctx context.Context, syncCtx factory.SyncContex
 		return nil
 	}
 
+	// Make sure that revision-ready is set to true on the
+	// revision-status{status.latestAvailableRevision} configmap. This prevents
+	// blocking the controller when upgrading to a version that checks revision
+	// readiness.
+	configMap, err := c.configMapGetter.ConfigMaps(c.targetNamespace).Get(ctx, nameFor("revision-status", operatorStatus.LatestAvailableRevision), metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if configMap.Annotations["operator.openshift.io/revision-ready"] != "true" {
+		configMap.Annotations["operator.openshift.io/revision-ready"] = "true"
```

Comment:
Shouldn't happen because this controller should be the only writer and should be creating these with at least one annotation, but it would panic here if configMap.Annotations is somehow nil.